### PR TITLE
Changed OS to ubuntu and added LibUv.

### PR DESCRIPTION
**Pull request by sebfia**
Additionally changed mono version to latest 4.2 stack which I found to be a lot stabler than 4.0 when it comes to threading.

### CODE REVIEW

**Comment by dsyme** [comment]
This base container shouldn't include libuv should it?  It's meant to be the base for all F# docker containers - the one from which all ones are derived and the "official" base docker container for F# - shouldn't libuv be added in a derived container?

**Comment by haf** [comment]
We could create a 'web' container; how would that be?

**Comment by sebfia** [comment]
Agreed! We should keep the mono 4.2.x stack though, as it is a lot stabler than 4.0.

**Comment by sebfia** [comment]
Btw.: Mono 4.2.1 can be regarded as stable from today on. http://www.mono-project.com/docs/about-mono/releases/4.2.1/

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,37 @@
...
 
-RUN set -x && \
-    yum clean all && \
-    yum install -y yum-utils && \
-    rpm --import "http://keyserver.ubuntu.com/pks/lookup?op=get&search=0x3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF" && \
-    yum-config-manager --add-repo http://download.mono-project.com/repo/centos/ && \
-    yum install -y mono-complete-$MONO_VERSION fsharp-$FSHARP_VERSION
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF && \
+    echo "deb http://download.mono-project.com/repo/debian wheezy main" | sudo tee /etc/apt/sources.list.d/mono-xamarin.list && \
+    echo "deb http://download.mono-project.com/repo/debian alpha main" | sudo tee /etc/apt/sources.list.d/mono-xamarin-alpha.list && \
```
Comment (naveensrinivasan) [inline Dockerfile:14]: Now that mono 4.2.1.102 is out do we still need the `alpha` ?
https://github.com/mono/docker/blob/master/4.2.1.102/Dockerfile

**Comment by haf** [comment]
@sebfia should you update your PR or should I?

**Comment by sebfia** [comment]
Sorry for the late reply!
I can do it.
Btw.: We could build the latest F# release from source since there is currently no package for it. For maintenance this would not make any difference as only the FSHARP_VERSION env. var. would need to be touched...

**Comment by sebfia** [comment]
No LibUv any more and we have F# 4.0.0.4

**Comment by haf** [comment]
Thanks, I'm testing this. Replacing maintainer with my own account though, but in no way diminishing your contribution, for which we're greatful!

**Comment by sebfia** [comment]
Of course - no problem!

**Pull request merged**